tasks/prepare-fbc-release/prepare-fbc-release.yaml

```diff
@@ -0,0 +1,114 @@
+      image: quay.io/hacbs-release/release-utils:a918b3a7cbd40add5955a2f1acb4c3dbb29c44e9
+      script: |
+        #!/usr/bin/env bash
+        set -eu
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+        if [ ! -f "${SNAPSHOT_PATH}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi        
+        
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        # Verify data.json exists
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+        fi        
+
+        pattern="^v[0-9]+\.[0-9]+$"
+
+        # Read components and initial values
+        components=$(jq -c '.components[]' "$SNAPSHOT_PATH")
+        fromIndex=$(jq -r '.fbc.fromIndex' "$DATA_FILE")
+        targetIndex=$(jq -r '.fbc.targetIndex' "$DATA_FILE")
+        binaryImage=$(jq -r '.fbc.binaryImage' "$DATA_FILE")
+
+        # Print initial values
+        echo "Initial fromIndex: $fromIndex"
+        echo "Initial targetIndex: $targetIndex"
+        echo "Initial binaryImage: $binaryImage"
+        echo
+
+        # Count components
+        COMPONENT_COUNT=$(jq '.components | length' "$SNAPSHOT_PATH")
+        echo "found $COMPONENT_COUNT components"
+
+        # Initialize jq_script
+        jq_script="."
+
+        # Function to replace tag in an image
+        replace_tag() {
+            local updatedImage="${1%:*}:${2}"
+            echo "$updatedImage"
+        }
+
+        # Iterate over components
+        while read -r component; do
+            containerImage=$(echo "$component" | jq -r '.containerImage')
+            componentName=$(echo "$component" | jq -r '.name')
+
+            # Fetch OCP version using skopeo
+            ocpVersion=$(skopeo inspect --raw docker://"$containerImage" \
+                | jq -r '.annotations."org.opencontainers.image.base.name"' | cut -d: -f2)
+
+            # Check if the version matches the pattern
+            if ! echo "$ocpVersion" | grep -Eq "$pattern"; then
+                echo "Invalid format for image $containerImage."
+                continue
+            fi
+
+            # Compute updated values
+            updatedFromIndex=$(replace_tag "$fromIndex" "$ocpVersion")
+            updatedTargetIndex=$(replace_tag "$targetIndex" "$ocpVersion")
+            updatedBinaryImage=$(replace_tag "$binaryImage" "$ocpVersion")
+
+            # Print updated values
+            echo "Component: $componentName"
+            echo "ocpVersion: $ocpVersion"
+            echo "Updated fromIndex for $componentName: $updatedFromIndex"
+            echo "Updated targetIndex for $componentName: $updatedTargetIndex"
+            echo "Updated binaryImage for $componentName: $updatedBinaryImage"
+            echo
+
+            # Build jq script to update JSON
+            jq_script+=" | (.components[] | select(.name == \"$componentName\") | .ocpVersion) |= \"$ocpVersion\""
```

Comment:
Can we rely on the component name being unique? I would actually suggest to run `jq` for each of the changes separately. And use the index instead of selecting based on name - you will have the array index if you switch to a for loop over component indexes.

Reply:
I agree with your point, fixed thanks. PTAL

Reply:
OK, cool. But I would still suggest to run the changes one by one. Have you considered it? For example, if something goes wrong, it will be much easier to debug which change broke - rather than seeing that this huge query failed.